examples/cpi.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019-2020 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use bcc::perf_event::{Event, HardwareEvent, SoftwareEvent};
+use bcc::BccError;
+use bcc::{PerfEvent, PerfEventArray, BPF};
+use clap::{App, Arg};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::{thread, time};
+
+// Both consants are arbitrary
+const DEFAULT_SAMPLE_FREQ: u64 = 99; // Hertz
+const DEFAULT_DURATION: u64 = 10; // Seconds
+
+fn do_main(runnable: Arc<AtomicBool>) -> Result<(), BccError> {
+    let matches = App::new("cpi")
+        .arg(
+            Arg::with_name("duration")
+                .long("duration")
+                .short("d")
+                .help("How long to run this trace for (in seconds)")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let duration: u64 = matches
+        .value_of("duration")
+        .map(|v| v.parse().expect("Invalid duration"))
+        .unwrap_or(DEFAULT_DURATION);
+
+    let cpus = bcc::cpuonline::get()?.len() as u32;
+
+    let code = format!(
+        "{}\n{}",
+        format!("#define NUM_CPU {}", cpus),
+        include_str!("cpi.c").to_string()
+    );
+
+    let mut bpf = BPF::new(&code)?;
+    PerfEventArray::new()
+        .table("instr_perf")
+        .event(Event::Hardware(HardwareEvent::Instructions))
+        .attach(&mut bpf)?;
+    PerfEventArray::new()
+        .table("cycle_perf")
+        .event(Event::Hardware(HardwareEvent::RefCpuCycles))
```

Comment:
Can we add fall-back to `HardwareEvent::CpuCycles`, possibly with a warning to the user to confirm the behavior for their microarchitecture wrt frequency scaling?